Automattic/wp-calypso
# Stats: Fix possible undefined value

Related to 
- https://a8c.sentry.io/issues/5024980638/events/04fd7b73d97644988e51d9bb31027355/
- https://a8c.sentry.io/issues/5803807699/events/8d1c44eb012345aaa8a664b68a035f1c/
- https://a8c.sentry.io/issues/5995127500/events/3fb772d5a9804482b454e28815e4e4d7/

<img width="1187" alt="image" src="https://github.com/user-attachments/assets/27ee3703-f81d-4f0e-90db-1216342d6cff">


## Proposed Changes

* Use conditional chaining for possibly undefined value

## Why are these changes being made?


* Bugfix

## Testing Instructions



* Open Live Branch, and ensure this URL doesn't crash the app `/stats/day/posts/:siteSlug?startDate=2024-10-23&summarize=1&num=2` 

## Pre-merge Checklist
